l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -401,6 +426,16 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         return (_chainId == eraChainId) && (_l2BatchNumber < eraFirstPostUpgradeBatch);
     }
 
+    /// @dev Determines if a deposit was initiated on zkSync Era before the upgrade to the Shared Bridge.
+    /// @param _chainId The chain ID of the transaction to check.
+    /// @param _l2BatchNumber The L2 batch number for the deposit where it was processed.
+    /// @param _l2TxNumberInBatch The L2 transaction number in the batch, in which the deposit was processed.
+    /// @return Whether deposit was initiated on zkSync Era before Shared Bridge upgrade.
+    function _isEraLegacyDeposit(uint256 _chainId, uint256 _l2BatchNumber, uint256 _l2TxNumberInBatch) internal view returns (bool) {
+        // Note: if eraLastOldBridgeDepositBatch or eraLastOldBridgeDepositTxNumber are 0, then this returns false, so normal security measures are applied
+        return (_chainId == eraChainId) && (_l2BatchNumber < eraLastOldBridgeDepositBatch) && (_l2TxNumberInBatch < eraLastOldBridgeDepositTxNumber);
```

Comment:
Should check that
`_chainId == eraChainId` &&
`_l2BatchNumber < eraLastOldBridgeDepositBatch || (_l2TxNumberInBatch < eraLastOldBridgeDepositTxNumber && _l2BatchNumber == eraLastOldBridgeDepositBatch)`

Otherwise older batches where tx happened to be in `eraLastOldBridgeDepositTxNumber` tx wouldn't work

Reply:
Very true